lib/puma/server.rb

```diff
@@ -116,12 +119,22 @@ def uncork_socket(socket)
         rescue IOError, SystemCallError
         end
       end
+
+      def closed_socket?(socket)
+        tcp_info = socket.getsockopt(Socket::SOL_TCP, Socket::TCP_INFO)
+        state = tcp_info.unpack("C".freeze)[0]
```

Comment:
Please pull that "C" out into a constant, as puma doesn't use the frozen string pragma.